### PR TITLE
feat: add private-to-private docs

### LINKED ISSUE #302: Populate "WebRTC Browser-to-Browser" in transport/webrtc
## Context

The libp2p maintainers have continued specifying the WebRTC specification and are adding support for browser-to-browser connectivity:

- The `Working Draft` version of the specification is in progress: https://github.com/libp2p/specs/pull/497.
- Per the working draft, three libp2p implementations are estimated to add support during Q2 2023 ([go-libp2p](https://github.com/libp2p/go-libp2p/blob/master/ROADMAP.md#end-of-q4-december), [js-libp2p](https://github.com/libp2p/js-libp2p/blob/master/ROADMAP.md#end-of-q4-december), and [rust-libp2p](https://github.com/libp2p/rust-libp2p/blob/master/ROADMAP.md#webrtc-support-browser-to-server)).
- [WebRTC interoperability testing](https://github.com/libp2p/test-plans/blob/master/ROADMAP.md#mid-q4-november) across implementations is also started mid Q4.

## Done Criteria

- [ ] Populate the "Browser-to-Browser" section in the WebRTC transport document.
- [ ] Audit and edit the original WebRTC transport document to ensure the general content does not revolve browser-to-server.
- [ ] Add a short comparison sub-section comparing the differences in connectivity
 
## Why Important

- Expanding the ways libp2p enables connectivity (into browser connectivity) is a primary goal of the libp2p project, WebRTC is centre to that. As we finish specification and implementation, we need corresponding documentation for the transport and it's use-cases.
- This continues the transport documentation for WebRTC started by #211.

### PR DESCRIPTION
**Pull request by p-shahi**
closes #302

### CODE REVIEW

**Comment by p-shahi** [comment]
@mxinden let me know what you think. Reading the specs was quite helpful but I may have gotten a few things wrong.

While this is ready for review, I think we should block this until both the implementation and specs PR are merged (I would like to link out to both from this doc.)

File: content/concepts/transports/webrtc.md
Comment (marten-seemann) [inline content/concepts/transports/webrtc.md:106]: We don't use the wasm compilation for browser to browser, nor elsewhere, do we? Consider removing?
```suggestion
Thanks to js-libp2p and rust-libp2p (compiled to Wasm), libp2p can run in the browser environment.
```

File: content/concepts/transports/webrtc.md
Comment (marten-seemann) [inline content/concepts/transports/webrtc.md:111]: They also can't dial TCP and QUIC connections.

File: content/concepts/transports/webrtc.md
Comment (marten-seemann) [inline content/concepts/transports/webrtc.md:114]: Why W3C? All the W3C is does is specify the browser API. We don't call WebTransport "WebTransport W3C".
```suggestion
Thankfully, libp2p solves this problem and enables browser node to browser node connectivity by supporting WebRTC.
```

File: content/concepts/transports/webrtc.md
Comment (marten-seemann) [inline content/concepts/transports/webrtc.md:117]: Not really. They're private nodes because people run their browsers in their home / corporate networks, and those networks almost always use NATs. In principle, you could run your browser on a machine that has a public address.

File: content/concepts/transports/webrtc.md
Comment (marten-seemann) [inline content/concepts/transports/webrtc.md:119]: Not 100% sure if this is correct.
```suggestion
This WebRTC solution enables connectivity between two non-browser nodes behind a NAT / firewall as well
```

File: content/concepts/transports/webrtc.md
Comment (marten-seemann) [inline content/concepts/transports/webrtc.md:124]: I don't think mentioning W3C makes a lot of sense here. I will engage on the specs PR. Browser-to-server also uses the W3C-defined API, simply because that's the API that's available in browsers.

File: content/concepts/transports/webrtc.md
Comment (marten-seemann) [inline content/concepts/transports/webrtc.md:126]: Same here, I don't think we should go with this multiaddr, but I'll comment on the PR in the multiaddr repo.

File: content/concepts/transports/webrtc.md
Comment (marten-seemann) [inline content/concepts/transports/webrtc.md:128]: It kind of is. The W3C gives you the signaling messages, you just need to figure out how to ship them to the peer yourself.

File: content/concepts/transports/webrtc.md
Comment (marten-seemann) [reply content/concepts/transports/webrtc.md:126]: Commented: https://github.com/multiformats/multiaddr/pull/150#event-8661620493.

File: content/concepts/transports/webrtc.md
Comment (p-shahi) [reply content/concepts/transports/webrtc.md:106]: My intention with this line was to indicate it's possible to run a rust libp2p node in the browser (when compiled to Wasm), not to indicate browser to browser is supported in rust-libp2p. I can either make this more clear or delete this.

File: content/concepts/transports/webrtc.md
Comment (p-shahi) [reply content/concepts/transports/webrtc.md:117]: I'll make this clearer

File: content/concepts/transports/webrtc.md
Comment (p-shahi) [reply content/concepts/transports/webrtc.md:114]: Leaving this until the convo in specs is resolved.

File: content/concepts/transports/webrtc.md
Comment (p-shahi) [reply content/concepts/transports/webrtc.md:128]: I see, might have misrepresented what was described [here](https://web.dev/webrtc-basics/#signaling-session-control,-network,-and-media-information): "Signaling methods and protocols are not specified by WebRTC. Signaling is not part of the RTCPeerConnection API."

I'll take a note out of this: https://webrtc.org/getting-started/peer-connections#signaling and reword as:
"Crucially, the WebRTC specification doesn't prescribe any single solution for signaling. The peer connection API has a notion of [signaling state](https://w3c.github.io/webrtc-pc/#dom-rtcsignalingstate) but not mechanism to share this data with remote peers."

lmk what you think

File: content/concepts/transports/webrtc.md
Comment (mxinden) [reply content/concepts/transports/webrtc.md:106]: It is not implemented yet. It is on the roadmap only.

https://github.com/libp2p/rust-libp2p/blob/master/ROADMAP.md#webrtc-support-browser-to-browser

File: content/concepts/transports/webrtc.md
Comment (mxinden) [inline content/concepts/transports/webrtc.md:128]:
```suggestion
However, there's more to p2p connections than what `RTCPeerConnection` provides. Crucially, signaling isn't built into the WebRTC API.
```

File: content/concepts/transports/webrtc.md
Comment (mxinden) [inline content/concepts/transports/webrtc.md:151]: _B_ will most likely not advertise its address through _R_ to _A_. This is a chicken-and-egg problem. How would _A_ know to connect to _R_ without _B_'s multiaddr?

```suggestion
- _B_ connects to _R_ and makes a relay reservation. (See circuit relay v2 docs.)
- _B_ appends `webrtc-w3c` to its relayed multiaddress and advertises it to the network.
- _A_ will discover _B_'s relayed multiaddr.
- _A_ sees that _B_ supports `webrtc-w3c` and establishes a relayed connection to _B_ via _R_.
```

File: content/concepts/transports/webrtc.md
Comment (mxinden) [inline content/concepts/transports/webrtc.md:153]: Just to avoid confusion, one does not create an `RTCPeerConnection` over the relayed connection, but rather with the help of the relayed connection. Is the phrasing only confusing me?

File: content/concepts/transports/webrtc.md
Comment (mxinden) [inline content/concepts/transports/webrtc.md:154]: I am confused. One can create an SDP offer via the `RTCPeerConnection` and then send that offer via the libp2p signaling stream. I don't know how one would send an SDP offer via the `RTCPeerConnection`.

File: content/concepts/transports/webrtc.md
Comment (mxinden) [inline content/concepts/transports/webrtc.md:146]: Is this step by step documentation needed? To me this is just rephrasing what is written in the spec without being higher level, thus more helpful to newcomers.

**Review by mxinden — COMMENTED** [review]
Thanks for the write-up @p-shahi. A couple of comments and suggestions.

File: content/concepts/transports/webrtc.md
Comment (p-shahi) [reply content/concepts/transports/webrtc.md]: Good point, I initially wrote the step by step to help me understand the flow but I agree it is a rehash. I rewrote it to keep it at a higher level and elide some of the details. Let me know what you think.

File: content/concepts/transports/webrtc.md
Comment (mxinden) [inline content/concepts/transports/webrtc.md:157]: The ICE candidates are part of the SDP, thus is a duplicate of the bullet above.

File: content/concepts/transports/webrtc.md
Comment (mxinden) [inline content/concepts/transports/webrtc.md:158]:
```suggestion
- The SDP messages are passed to the browsers WebRTC stack, which then tries to establish a direct connection.
```

File: content/concepts/transports/webrtc.md
Comment (mxinden) [inline content/concepts/transports/webrtc.md:164]: Is this really necessary for newcomers to know?

**Review by mxinden — COMMENTED** [review]
Couple smaller nits. Overall this looks good to me. Thanks for the follow-ups.

File: content/concepts/transports/webrtc.md
Comment (p-shahi) [inline content/concepts/transports/webrtc.md:164]:
```suggestion
```

File: content/concepts/transports/webrtc.md
Comment (p-shahi) [inline content/concepts/transports/webrtc.md:157]:
```suggestion
    - Exchange SDPs between nodes (an offer from _A_ and an answer from _B_)
      - This include exchangning information about the network connection (in WebRTC parlance this is called [exchanging ICE candidates](https://developer.mozilla.org/en-US/docs/Web/API/WebRTC_API/Connectivity#ice_candidates))
```

File: content/concepts/transports/webrtc.md
Comment (p-shahi) [reply content/concepts/transports/webrtc.md]: Removed.

File: content/concepts/transports/webrtc.md
Comment (p-shahi) [reply content/concepts/transports/webrtc.md]: I would like to keep this, made it a subpoint of the previous line.

**Review by mxinden — APPROVED** [review]
Thanks for the work here.

**Comment by p-shahi** [comment]
Thanks @mxinden will wait for either https://github.com/libp2p/js-libp2p-webrtc/pull/90 or https://github.com/libp2p/specs/pull/497 to land before merging

**Comment by p-shahi** [comment]
https://github.com/libp2p/js-libp2p-webrtc/pull/90 is in, merging!

**Pull request merged**
